### PR TITLE
Add auth page

### PR DESCRIPTION
**Pull request by firehawk89**
For now I've just added an auth page as well as created some components which would be used there

### CODE REVIEW

File: src/app/auth/page.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export default function Auth() {
+  return (
+    <section className="h-screen w-screen flex items-center justify-center bg-gradient-blue-white">
+      <div className="card p-10 flex gap-[3.75rem] bg-white">
+        <ImageCard className="-my-[4.375rem] flex-shrink-0">
+          <Image alt="Golub" height={161} src="/images/golub.png" width={196} />
+        </ImageCard>
+        <div>
+          <h1 className="title-lg">Authentication</h1>
+          <AuthForm />
```
Comment (DemianParkhomenko) [inline src/app/auth/page.tsx:14]: Very subjectively, but I would name it `FormAuth` instead to make a common prefix at the beginning. Then all forms will start with `Form*` prefix. 

P.S. the same as we have `IconCheck` now

File: src/components/Pages/Auth/AuthForm.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const AuthForm: FC = () => {
+  return (
+    <Form
+      className="mt-5"
+      onSubmit={(e: FormEvent<HTMLFormElement>) => console.log(e.target)}
+    >
+      <FormControl>
+        <FormLabel htmlFor="email">
+          Confirm your email and get dynamically generated code
```
Comment (DemianParkhomenko) [inline src/components/Pages/Auth/AuthForm.tsx:19]: We did not discuss it yet, but I think we should add i18n. You can skip it for this PR, and we will refactor later.

File: src/components/UI/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { FC, ReactNode } from 'react'
+
+type ButtonProps = {
+  children: ReactNode
+  className?: string
+  disabled?: boolean
+  type?: 'button' | 'reset' | 'submit'
+}
+
+const THEMES = {
```
Comment (DemianParkhomenko) [inline src/components/UI/Button.tsx:11]: Should a common type be here too?

File: src/components/UI/Icons/IconCheck.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import IconProps from '@/types/IconProps'
+import { ICON_SIZE } from '@/types/enums/IconSize'
+import { FC } from 'react'
+
+const IconCheck: FC<IconProps> = ({ className, size }) => {
+  const dimension = size ? ICON_SIZE[size] : ICON_SIZE['md']
```
Comment (DemianParkhomenko) [inline src/components/UI/Icons/IconCheck.tsx:6]: Seems this line will be always copy-pasted and it is bad.

**Review by DemianParkhomenko — CHANGES_REQUESTED** [review]
👍

File: src/app/auth/page.tsx (hunk 1 above)
Comment (firehawk89) [reply src/app/auth/page.tsx:14]: OK, that makes sense

File: src/components/Pages/Auth/AuthForm.tsx (hunk 2 above)
Comment (firehawk89) [reply src/components/Pages/Auth/AuthForm.tsx:19]: Surely, for now it's just for testing purposes

File: src/components/UI/Button.tsx (hunk 3 above)
Comment (firehawk89) [reply src/components/UI/Button.tsx:11]: You mean type for THEMES?

File: src/components/UI/Icons/IconCheck.tsx (hunk 4 above)
Comment (firehawk89) [reply src/components/UI/Icons/IconCheck.tsx:6]: For now I'll just create a misc utility for getting the dimension

File: public/images/golub.png
[the remark this answers is not in the document]
Comment (firehawk89) [reply public/images/golub.png:1]: Not really, it's just an image which has no effects on hover etc., so svg is not necessary

File: src/components/UI/Button.tsx (hunk 3 above)
Comment (DemianParkhomenko) [reply src/components/UI/Button.tsx:11]: yes

Comment (firehawk89) [reply src/components/UI/Button.tsx:11]: Will do this when implementing theme functionality

**Review by DemianParkhomenko — APPROVED** [review]
👍 Nice one

File: src/app/globals.css [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+.caption {
+  @apply text-xs uppercase;
+}
+
+.card {
+  @apply rounded-3xl;
+}
+
+.input-placeholder {
+  @apply peer-[:not(:placeholder-shown)]:translate-y-0 peer-[:not(:placeholder-shown)]:top-[0.375rem] peer-[:not(:placeholder-shown)]:text-[0.6875rem] peer-[:not(:placeholder-shown)]:leading-[0.8125rem];
```
[the remark this answers is not in the document]
Comment (firehawk89) [reply src/app/globals.css:38]: Not possible

File: src/components/UI/Container.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { classnames } from '@/utils'
+import { FC, ReactNode } from 'react'
+
+type ContainerProps = {
+  children: ReactNode
+  className?: string
+}
+
+const Container: FC<ContainerProps> = ({ children, className }) => {
```
[the remark this answers is not in the document]
Comment (firehawk89) [reply src/components/UI/Container.tsx:9]: Renamed to Layout

File: src/components/UI/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  dark: 'disabled:bg-none disabled:text-base-gray-8 disabled:bg-base-gray-7',
+  light: 'disabled:bg-none disabled:text-base-gray-4 disabled:bg-base-gray-3',
+}
+
+const Button: FC<ButtonProps> = ({ children, className, disabled, type }) => {
+  return (
```
[the remark this answers is not in the document]
Comment (firehawk89) [reply src/components/UI/Button.tsx:17]: Will work with form logic in the next PR, because this one will be too large

**Review by dEdmishka — APPROVED** [review]
Cool

**Pull request merged**